src/sagemaker/fw_utils.py

```diff
@@ -747,12 +764,56 @@ def validate_distribution(
             distribution=distribution,
             image_uri=image_uri,
         )
+        validate_pytorch_distribution(
+            framework_name=framework_name,
+            framework_version=framework_version,
+            py_version=py_version,
+            image_uri=image_uri,
+        )
         warn_if_parameter_server_with_multi_gpu(
             training_instance_type=instance_type, distribution=distribution
         )
     return distribution
 
 
+def validate_pytorch_distribution(framework_name, framework_version, py_version, image_uri):
+    """Check if pytorch distribution strategy is correctly invoked by the user.
+
+    Args:
+        framework_name (str): A string representing the name of framework selected.
+        framework_version (str): A string representing the framework version selected.
+        py_version (str): A string representing the python version selected.
+        image_uri (str): A string representing a Docker image URI.
+
+    Raises:
+        ValueError: if
+            `py_version` is not python3 or
+            `framework_version` is not in PYTORCHDDP_SUPPORTED_FRAMEWORK_VERSIONS
+    """
+    if framework_name != "pytorch":
+        # We need to validate only for PyTorch framework
+        return
+
+    err_msg = ""
+    if not image_uri:
```

Comment:
what will happen if using an custom docker image uri and there is old toolkit installed there?

Reply:
I am mirroring the logic that we have for smdistributed: https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/fw_utils.py#L635.
How can we check for older toolkit versions inside custom DLCs?